### ISSUE TITLE
Homepage Posts: Optimize image sizes (after #543)

### ISSUE THREAD

**Issue by dkoo** [Automattic/newspack-blocks#544]
This issue is related to #543 and only applies after it takes effect. Filing it as a separate issue so we don't forget, but also don't want to complicate that PR any more than we have to.

The Homepage Posts block attempts to use Photon to resize featured images based on the block's attributes (portrait, landscape, or square). If Photon is not available (because Jetpack is not activated or installed), it falls back to WP's default Large image size (1024px max).

Currently, all images displayed by the block will display up to 1200px (or up to the original image size) in all contexts and all breakpoints/devices. The default WP `srcset` and `sizes` attributes are removed intentionally because they resulted in loss of fidelity due to soft-cropping images via CSS to fit the aspect ratio selected in the block attributes, so these images are no longer responsive.

Still, I think there's some room for further optimization here by taking a breakpoint-oriented approach. 

* For Photon-resized images, we can manually build `sizes` and `srcset` attributes based on the theme's breakpoints.
* For fallback WP images, we might want to stick with Large because Medium (300px) is likely too small to render properly on all but the smallest devices. But maybe worth investigating using sizes and srcset based on Large and Medium, and testing visual degradation vs. performance gain on mobile devices.
* Most Newspack sites will have Jetpack enabled, so I don't see the fallback being used in many cases.
* This approach could be potentially fine-tuned further to add sizes and srcset URLs based on potential container sizes, but that would complicate the task a lot.
